src/SourceTerms/Deleptonization/CPU_Src_Deleptonization.cpp

```diff
@@ -124,9 +145,138 @@ static void Src_Deleptonization( real fluid[], const real B[],
+         real Ye        = fluid[YE] / fluid[DENS];
+
+   real Eint_Code_0 = Eint_Code;
+   real Entr_0      = Entr;
+   real Ye_0        = Ye;
+
+
+   if ( Dens_CGS <= Delep_minDens_CGS )
+   {
+      Del_Ye = 0.0;
+   } else
+   {
+      Yout = YeOfRhoFunc( Dens_CGS, DELEP_RHO1, DELEP_RHO2,
+                          DELEP_YE1, DELEP_YE2, DELEP_YEC );
+      Del_Ye = Yout - Ye;
+      Del_Ye = MIN( 0.0, Del_Ye ); // Deleptonization cannot increase Ye
+   }
+
+   if ( Del_Ye < 0.0 )
+   {
+//    Nuclear EoS
+#     if ( NUC_TABLE_MODE == NUC_TABLE_MODE_TEMP )
+      const int  NTarget1 = 2;
+#     else
+      const int  NTarget1 = 3;
+#     endif
+            int  In_Int1[NTarget1+1];
+            real In_Flt1[3], Out1[NTarget1+1];
+
+      In_Flt1[0] = Dens_Code;
+      In_Flt1[1] = Eint_Code;
+      In_Flt1[2] = Ye;
+
+      In_Int1[0] = NTarget1;
+      In_Int1[1] = NUC_VAR_IDX_ENTR;
+      In_Int1[2] = NUC_VAR_IDX_MUNU;
+#  if ( NUC_TABLE_MODE == NUC_TABLE_MODE_ENGY )
+      In_Int1[3] = NUC_VAR_IDX_EORT;
+#  endif
+
+
+      EoS->General_FuncPtr( NUC_MODE_ENGY, Out1, In_Flt1, In_Int1, EoS->AuxArrayDevPtr_Flt, EoS->AuxArrayDevPtr_Int, EoS->Table );
+
+                 Entr      = Out1[0];
+            real mu_nu_MeV = Out1[1];
+      const real Temp_MeV  = Out1[2] * Kelvin2MeV;
+
+#  ifdef GAMER_DEBUG
+   if ( mu_nu_MeV != mu_nu_MeV ) printf( "ERROR : Couldn't get chemical potential munu (NaN) !!\n" );
+#  endif // GAMER_DEBUG
+
+      if ( ( mu_nu_MeV < DELEP_ENU ) || ( Dens_CGS >= 2.0e12 ) )
+      {
+         Del_Entr = 0.0;
+      } else
+      {
+         Del_Entr = - Del_Ye * ( mu_nu_MeV - DELEP_ENU ) / Temp_MeV;
+      }
+
+//    update entropy and Ye
+      Entr      = Entr + Del_Entr;
+      Ye        = Ye + Del_Ye;
+      fluid[YE] = Dens_Code * Ye;
+
+
+//    input and output arrays for Nuclear EoS
+#     if ( NUC_TABLE_MODE == NUC_TABLE_MODE_TEMP )
+      const int  NTarget2 = 1;
+#     else
+      const int  NTarget2 = 0;
+#     endif
+            int  In_Int2[NTarget2+1];
+            real In_Flt2[3], Out2[NTarget2+1];
+
+      In_Flt2[0]  = Dens_Code;
+      In_Flt2[1]  = Entr;
+      In_Flt2[2]  = Ye;
+
+      In_Int2[0]  = NTarget2;
+#     if ( NUC_TABLE_MODE == NUC_TABLE_MODE_TEMP )
+      In_Int2[1]  = NUC_VAR_IDX_EORT;
+#     endif
+
+//    call Nuclear EoS with entropy mode
+      EoS->General_FuncPtr( NUC_MODE_ENTR, Out2, In_Flt2, In_Int2, EoS->AuxArrayDevPtr_Flt, EoS->AuxArrayDevPtr_Int, EoS->Table );
+      const real Eint_Update = Out2[0];
+      fluid[ENGY] = Hydro_ConEint2Etot( fluid[DENS], fluid[MOMX], fluid[MOMY], fluid[MOMZ], Eint_Update, Emag );
+
+
+
+// final check
+#  if GAMER_DEBUG
+   if (  Hydro_CheckUnphysical( UNPHY_MODE_SING, &Eint_Update, "output internal energy density", ERROR_INFO, UNPHY_VERBOSE )  )
+   {
+      printf( "   Dens=%13.7e code units, Eint=%13.7e code units, Ye=%13.7e\n", Dens_Code, Eint_Code, Ye );
+   }
+#  endif // GAMER_DEBUG
```

Comment:
Update the indentation, and probably more information for debugging (Del_Ye, etc).

Reply:
added `Entr`, `Del_Ye` and `Del_Entr`.

Reply:
Since you move this check outside the `if ( Del_Ye < 0.0 ) ... else` block, we now need to assing the initail values of `Entr`, `Del_Ye` and `Del_Entr` to `NULL_REAL`

Reply:
You're right. I updated this.